mysql-test/t/percona_dd_upgrade_undo_encrypted.test

```diff
@@ -0,0 +1,153 @@
+call mtr.add_suppression("ACL table mysql.[a-zA-Z_]* missing. Some operations may fail.");
+call mtr.add_suppression("Info table is not ready to be used. Table 'mysql.slave_master_info' cannot be opened");
+call mtr.add_suppression("Error in checking mysql.slave_master_info repository info type of TABLE");
+call mtr.add_suppression("Error creating master info: Error checking repositories.");
+call mtr.add_suppression("Slave: Failed to initialize the master info structure for channel");
+call mtr.add_suppression("Failed to create or recover replication info repositories.");
+call mtr.add_suppression("db.opt file not found for test database. Using default Character set");
+call mtr.add_suppression("Skip re-populating collations and character sets tables in InnoDB read-only mode");
+call mtr.add_suppression("Skipped updating resource group metadata in InnoDB read only mode");
+--enable_query_log
+
+--echo # Set different paths for --datadir
+let $MYSQLD_DATADIR1 = $MYSQL_TMP_DIR/undo57_encrypted/datadir/data;
+
+--echo # Set path for undo* files.
+let $MYSQLD_UNDO_DATADIR = $MYSQL_TMP_DIR/undo57_encrypted/innodb_undo_data_dir;
+
+--echo # Set path for ibdata* files.
+let $MYSQLD_HOME_DATA_DIR = $MYSQL_TMP_DIR/undo57_encrypted/innodb_data_home_dir;
+
+--copy_file $MYSQLTEST_VARDIR/std_data/undo57_encrypted.zip $MYSQL_TMP_DIR/undo57_encrypted.zip
+
+--echo # Check that the file exists in the working folder.
+--file_exists $MYSQL_TMP_DIR/undo57_encrypted.zip
+
+--echo # Unzip the zip file.
+--exec unzip -qo $MYSQL_TMP_DIR/undo57_encrypted.zip -d $MYSQL_TMP_DIR
+
+--echo # Stop DB server which was created by MTR default, empty undo logs
+SET GLOBAL innodb_fast_shutdown=0;
+--source include/shutdown_mysqld.inc
+
+--echo # Start the 8.0 server on 5.7 datadir with correct keyring_file
+--let $restart_parameters="restart: --early-plugin-load="keyring_file=$KEYRING_PLUGIN" $KEYRING_PLUGIN_OPT --keyring_file_data=$MYSQL_TMP_DIR/undo57_encrypted/my_key_undo5 --innodb_data_home_dir=$MYSQLD_HOME_DATA_DIR --innodb_undo_directory=$MYSQLD_UNDO_DATADIR --datadir=$MYSQLD_DATADIR1"
+--replace_result $MYSQL_TMP_DIR TMP_DIR $KEYRING_PLUGIN_OPT --plugin-dir=KEYRING_PLUGIN_PATH $MYSQLD_HOME_DATA_DIR HOME_DIR $MYSQLD_UNDO_DATADIR UNDO_DATADIR $MYSQLD_DATADIR1 DATADIR
+--source include/start_mysqld_no_echo.inc
+
+--echo # Execute mysql_upgrade
+
+--source include/mysql_upgrade_preparation.inc
+--exec $MYSQL_UPGRADE --skip-verbose --force 2>&1
+--source include/mysql_upgrade_cleanup.inc
+
+
+SHOW CREATE TABLE test.tab1;
+SELECT * FROM tab1;
+
+--source include/shutdown_mysqld.inc
+
+--echo # Now lets test what happens when encryption key is missing and we try to start 8.0 server on 5.7 directory
+--echo # with encrypted undo tablespaces
+--echo # Recreate the directories
+--force-rmdir $MYSQL_TMP_DIR/undo57_encrypted
+--echo # Check that the file exists in the working folder.
+--file_exists $MYSQL_TMP_DIR/undo57_encrypted.zip
```

Comment:
Remove

Reply:
I cannot remove the --force-rmdir part as there are some leftovers and server refuses to start:
2019-04-05T16:27:38.197914Z 1 [ERROR] [MY-012209] [InnoDB] Multiple files found for the same tablespace ID:
2019-04-05T16:27:38.197938Z 1 [ERROR] [MY-012202] [InnoDB] Tablespace ID: 15 = ['mysql/innodb_table_stats.ibd', 'mysql/innodb_table_stats_backup57.ibd']

2019-04-05T16:27:38.197955Z 1 [ERROR] [MY-012202] [InnoDB] Tablespace ID: 16 = ['mysql/innodb_index_stats.ibd', 'mysql/innodb_index_stats_backup57.ibd']

Reply:
No remove not the force-rmdir but file_exists line

Reply:
ack, this was done